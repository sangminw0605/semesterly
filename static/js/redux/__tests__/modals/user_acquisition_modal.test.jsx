/*
Copyright (C) 2017 Semester.ly Technologies, LLC

Semester.ly is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

Semester.ly is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
*/
import React from 'react';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import Provider from 'react-redux/src/components/Provider';
import configureMockStore from 'redux-mock-store';
import { userInfoFixture } from '../../__fixtures__/user_acquisition_modal.fixture';
import UserAcquisitionModalContainer from '../../ui/containers/modals/user_acquisition_modal_container';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('User Aquisition Modal', () => {
  it('shows when isVisible is true', () => {
    const store = mockStore({
      userAcquisitionModal: { isVisible: true },
      userInfo: userInfoFixture,
    });

    const { container } = render(
      <Provider store={store}>
        <UserAcquisitionModalContainer />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('is hidden when isVisible is false', () => {
    const store = mockStore({
      userAcquisitionModal: { isVisible: false },
      userInfo: userInfoFixture,
    });
    const { container } = render(
      <Provider store={store}>
        <UserAcquisitionModalContainer />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
