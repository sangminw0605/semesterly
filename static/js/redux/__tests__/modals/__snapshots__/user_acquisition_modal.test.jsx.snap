// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Aquisition Modal is hidden when isVisible is false 1`] = `<div />`;

exports[`User Aquisition Modal shows when isVisible is true 1`] = `
<div>
  <span>
    <div
      style="z-index: 1050; position: fixed; width: 100%; top: 50%; left: 50%;"
      class="user-acquisition-modal abnb-modal max-modal"
    >
      <div
        tabindex="-1"
        style="margin: 0px; background-color: white;"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <h1>
              Login/Signup
            </h1>
          </div>
        </div>
        <div
          class="user-acquisition-modal__container"
        >
          <h3>
            Recommended:
          </h3>
          <button
            class="btn abnb-btn fb-btn"
          >
            <span
              class="img-icon"
            >
              <i
                class="fa fa-facebook"
              />
            </span>
            <span>
              Continue with Facebook
            </span>
          </button>
          <p
            class="method-details"
          >
            Allows the option to friends in your classes.
          </p>
          <br />
          <button
            class="btn abnb-btn secondary"
          >
            <span
              class="img-icon"
            >
              <img
                alt="JHU"
                class="jhu-square"
                src="/static/img/school_logos/jhu-square.png"
              />
            </span>
            <span>
              Continue with JHED*
            </span>
          </button>
          <p
            class="method-details"
          >
            * Exclusive to JHU students & faculty.
          </p>
          <div
            class="or-separator"
          >
            <span
              class="h6 or-separator--text"
            >
              or
            </span>
            <hr />
          </div>
          <button
            class="btn abnb-btn secondary"
          >
            <span
              class="img-icon"
            >
              <img
                alt="Google"
                class="google-logo"
                src="https://a0.muscache.com/airbnb/static/signinup/google_icon_2x-745c2280e5004d4c90e3ca4e60e3f677.png"
              />
            </span>
            <span>
              Continue with Google
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      style="position: fixed; top: 0px; right: 0px; bottom: 0px; left: 0px; z-index: 1040; background-color: rgb(55, 58, 71);"
    />
  </span>
</div>
`;
